examples/solvers/pdhg_denoising_L1_HuberTV.py

```diff
@@ -0,0 +1,123 @@
+"""Total variation denoising using PDHG.
+
+Solve the L1-HuberTV problem
+
+        min_{x >= 0} ||x - d||_1 + lam TV_gamma(x)
+
+where ``grad`` the spatial gradient and ``d`` is given noisy data.
+
+For further details and a description of the solution method used, see
+https://odlgroup.github.io/odl/guide/pdhg_guide.html in the ODL documentation.
+"""
+
+import numpy as np
+import scipy
+import odl
+import matplotlib.pyplot as plt
+
+# --- define setting --- #
+
+# Read test image: use only every second pixel, convert integer to float
+image = np.rot90(scipy.misc.ascent()[::2, ::2].astype('float'), 3)
+shape = image.shape
+
+# Rescale max to 1
+image /= image.max()
+
+# Discretized spaces
+space = odl.uniform_discr([0, 0], shape, shape)
+
+# Create space element of ground truth
+orig = space.element(image.copy())
+
+# Add noise and convert to space element
+noisy = odl.phantom.salt_pepper_noise(orig)
+
+# Gradient operator
+gradient = odl.Gradient(space, method='forward')
+
+# regularization parameter
+reg_param = 1
+
+# l1 data matching
+l1_norm = 1 / reg_param * odl.solvers.L1Norm(space).translated(noisy)
+
+# HuberTV-regularization
+huber_l1_norm = odl.solvers.HuberL1L2(gradient.range, gamma=.1)
+
+# define objective
+obj_fun = l1_norm + huber_l1_norm * gradient
+
+# strong convexity of "f*"
+strong_convexity = 1 / huber_l1_norm.grad_lipschitz
+
+
+# define callback to store function values
+class CallbackStore(odl.solvers.util.callback.Callback):
```

Comment:
This adds *a lot* of complexity, are we truly sure we need it? I see why we do it, but I'm afraid it will scare off new users.

Reply:
I see your point. This is also related with your top comment about "advanced" examples. On the one hand, this may be too advanced for some beginners, but then this is something many people need in practice which would justify it being part of the examples. I would suggest to keep it for now, but to move this example somewhere else in the future.

Reply:
Sounds like a reasonable solution, perhaps also write something in the top level comment that this example is "exhaustive" and "not for beginners" so to say.